sktime/transformations/series/feature_selection.py

```diff
@@ -0,0 +1,187 @@
+    n_columns : int, optional
+        Number of feautres (columns) to select. n_columns must be <=
+        number of X columns. Some methods require n_columns to be given.
+    random_state : int/float/str, optional
+        Value to set random.seed() if method="random", default None
+        Requires n_columns to be given.
+    columns : list of str
+        A list of columns to select. If columns is given.
+
+    Attributes
+    ----------
+    columns_ : list of str
+        List of columns that have been selected as features.
+    regressor_ : sklearn-like regressor
+        Fitted regressor (meta-model).
+    n_columns_: int
+        Derived from number of features if n_columns is None, then
+        n_columns_ is calculated as int(math.ceil(Z.shape[1] / 2)). So taking
+        half of given features only as default.
+    feature_importances_ : dict or None
+        A dictionary with column name as key and feature imporatnce value as value.
+        The dict is sorted descending on value. This attribute is a dict if
+        method="feature-importances", else None.
+
+    Examples
+    --------
+    >>> from sktime.transformations.series.feature_selection import FeatureSelection
+    >>> from sktime.datasets import load_longley
+    >>> y, X = load_longley()
+    >>> transformer = FeatureSelection(method="feature-importances", n_columns=3)
+    >>> Xt = transformer.fit_transform(X, y)
+    """
+
+    _tags = {
+        "fit-in-transform": False,
+        "transform-returns-same-time-index": True,
+        "skip-inverse-transform": True,
+        "univariate-only": False,
+    }
+
+    def __init__(
+        self,
+        method="feature-importances",
+        n_columns=None,
+        regressor=None,
+        random_state=None,
+        columns=None,
+    ):
+        self.n_columns = n_columns
+        self.method = method
+        self.regressor = regressor
+        self.random_state = random_state
+        self.columns = columns
+
+        super(FeatureSelection, self).__init__()
+
+    def fit(self, Z, X=None):
+        """Fit the transformation on input series `Z`.
+
+        Parameters
+        ----------
+        Z : pd.DataFrame
+            A time series to apply the transformation on.
+
+        Returns
+        -------
+        self
+        """
+        Z = check_series(Z)
+        self.n_columns_ = self.n_columns
+        self.feature_importances_ = None
+
+        # multivariate Z
+        if not isinstance(Z, pd.Series):
+            if self.method == "feature-importances":
+                self._check_regressor()
+                self._check_n_columns(Z)
+                X = check_series(X)
+                # fit regressor with Z as exog data and X as endog data (target)
+                self.regressor_.fit(X=Z, y=X)
+                # create dict with columns name (key) and feauter importance (value)
+                d = dict(zip(Z.columns, self.regressor_.feature_importances_))
```

Comment:
What happens if `feature_importances_` isn't an attribute of the `self.regressor_`? I saw that you describe that in the docstring which is great but perhaps we could check for that an raise an error message?

Reply:
yes, let me add a check for this to raise a proper exception message